src/components/create-vm-dialog/createVmDialogUtils.js

```diff
@@ -145,3 +146,11 @@ export function correctSpecialCases(os) {
 export function needsRHToken(osName) {
     return osName.startsWith("rhel");
 }
+
+export function saveOfflineToken(offlineToken) {
+    return cockpit.user().then(user => localStorage.setItem(user.id + "-offline-token", offlineToken));
```

Comment:
Thinking aloud.. this uses the same name for all remote hosts. But that should be fine, the token is not dependent on a particular machine, and the browser user can use the same token for all of them. But the same is true for `cockpit.user()` -- if I log into a remote machine as two different Linux users, the human browser user can still use the same token. Hence I think this should drop the per-user storage here.

This also avoids cluttering the storage with an unbounded number of tokens, and this remains self-cleaning.

Reply:
I know its a niche scenario, but if multiple different users log thru the same computer, one will get an offline token of the other user.
Since the benefit of the cockpit is that you manage your server from anywhere, let's imagine scenario of 2 colleagues, or a scenario in some lab where user logs into cockpit with some public/shared client.
I know it's a niche scenario, but offline token is a very sensitive information, since with that you can manage users[ whole Red Hat subscription](https://access.redhat.com/management/api/rhsm), e.g. manage its systems, allocations, errata... And not just subscriptions, but also manage [accounts](https://access.redhat.com/management/api/account) and [cases](https://access.redhat.com/management/api/case_management)

Reply:
Different users on the same computer have separate user accounts and sessions, and thus also separate localStorages. If two humans literally use the same linux account, then (1) security-wise they *are* the same identity and share these secrets anyway, and (2) cockpit.user() will then be the same, so storing the tokens per-user name will not change anything here.

My point is, an offline token is *not* attached to a particular linux account on a particular machine. It's attached to the *browser* user. So it doesn't make sense to include the remote account name here.

Reply:
Agree with Martin. Also this avoids accessing of user name which I always found a bit awkward.